### ISSUE TITLE
Improvement of IPF library loading

### ISSUE THREAD

**Issue by sebhz** [ColinPitrat/caprice32#126]
This is more an improvement of an existing feature than a real issue.
We should use dlopen() and friends in the ipf.cpp file, rather than linking with the libcapsimage.so.

This would allow to package a debian release with IPF enabled without distributing libcapsimage.so, and keep the emulator functional if the user does not have IPF library installed on his/her system (and does not need IPF, of course).
As an added benefit, we could *always* enable IPF once this is done, thus making makefiles simpler.

This seems to be what the LoadLibrary windows call does.

**Comment by ColinPitrat** [comment]
At compilation time, we'll need the headers though (caps/*.h) so that it builds:
https://github.com/ColinPitrat/caprice32/blob/master/src/ipf.h

**Comment by sebhz** [comment]
Correct. We will still need the WITH_IPF flag for the build, just to allow people to build without IPF support if they do not care about IPF and do not want to download the IPF package.
